app/Http/Controllers/OperationController.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Comments;
+use App\CourseStudent;
+use App\User;
+use App\Http\Controllers\Controller;
+use App\Models\Course;
+
+class OperationController extends Controller
+{
+    /**
+     * Use to get operation code and output the correct outout.
+     * 
+     * @param int $courseId
+     * 
+     * @return Response
+     */
+    static public function show($courseId)
+    {
+        $ret = [];
+        
+        $course = Course::find($courseId);
+
+        if (!$course) {
+            abort(404);
+        }
+
+        $searchResult = $course->attendStudents;
+
+        $isSearchNotEmpty = !(count($searchResult) == 0);
+        $isSearchValid = ($course != null) ? true : false;
+
+        $comment = $course->comments;
+        $isCommentNotEmpty = ( count($comment) != 0) ? true : false;
+        
+        if ($isSearchValid) {
+            $ret[ 'searchResult' ] = $isSearchNotEmpty ? ($searchResult) : false;
+
+            $ret[ 'name' ] = $course->name;
+            $ret[ 'description'] = $course->description;
+            $ret[ 'outline' ] = $course->outline;
+            
+            $ret[ 'comments' ] = ($isCommentNotEmpty) ? $comment : false;
+
+            return view('operation_valid', $ret);
+        } else {
+            abort(404);
+        }
```

Comment:
可以 early return 減少巢狀
```suggestion
        if (! $isSearchValid) {
            abort(404);
        }
        $ret[ 'searchResult' ] = $searchResult ?? false;

        $ret[ 'name' ] = $course->name;
        $ret[ 'description'] = $course->description;
        $ret[ 'outline' ] = $course->outline;
            
        $ret[ 'comments' ] = ($isCommentNotEmpty) ? $comment : false;

        return view('operation_valid', $ret);
```

Reply:
- 已應用建議的內容